addons/main.go

```diff
@@ -378,11 +396,17 @@ func enableWebhooks(ctx context.Context, mgr ctrl.Manager, flags *addonFlags) {
 }
 
 func enablePackageInstallStatusController(ctx context.Context, mgr ctrl.Manager, flags *addonFlags) {
-	// set up a ClusterCacheTracker to provide to PackageInstallStatus controller which requires a connection to remote clusters
-	// the informers/caches are created only for objects accessed through Get/List in the code.
-	// we only read Package and PackageInstall resources through our cached client, we let those two resource types to be cached and we don't add any resource type to the ClientUncachedObjects list
+	// Exclude kapppkg.PackageInstall and kappdatapkg.Package from tracker caches to prevent too many items
+	// being held in memory along with standard ConfigMap and Secret
 	l := ctrl.Log.WithName("remote").WithName("ClusterCacheTracker")
-	tracker, err := capiremote.NewClusterCacheTracker(mgr, capiremote.ClusterCacheTrackerOptions{Log: &l})
+	tracker, err := capiremote.NewClusterCacheTracker(mgr, capiremote.ClusterCacheTrackerOptions{Log: &l,
+		ClientUncachedObjects: []client.Object{
+			&corev1.ConfigMap{},
+			&corev1.Secret{},
+			&kapppkg.PackageInstall{},
```

Comment:
Wouldn't adding ```PackageInstall``` & ```Package``` to ```ClientUncachedObjects```, mean that we'll not allow objects of those types to be cached anymore? Does that mean that we will be reading from api server for those now and that we'll be not using the cache functionality of remote watch anymore?